packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -144,13 +152,22 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
         }
     }
 
-    protected async newTerminal(): Promise<void> {
-        const widget = <TerminalWidget>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
-            created: new Date().toString()
+    public async newTerminal(options: TerminalWidgetOptions): Promise<TerminalWidget & Widget> {
+        const widget = <TerminalWidgetImpl>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
+            created: new Date().toString(),
+            ...options
         });
+        return widget;
+    }
+
+    public activateWidget(widget: TerminalWidget & Widget): void {
         this.shell.addWidget(widget, { area: 'bottom' });
         this.shell.activateWidget(widget.id);
-        widget.start();
     }
 
+    public collapseWidget(termWidget: TerminalWidget & Widget): void {
+        if (termWidget.isVisible) {
+            this.shell.collapsePanel('bottom');
```

Comment:
Is this always correct?

Reply:
Did you mean extension/plugin writer can create terminal with help TerminalService and attach widget manually to the another panel (for example right panel) and use collapseWidget method?

Reply:
Exactly.

Reply:
It looks strange that `collapseWidget` will collapse the whole bar. Where is this method used? What are the client expectations?